### PR TITLE
Unify API

### LINKED ISSUE #45: Unify API

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #45

### CODE REVIEW

File: sdk/src/main/java/com/oursky/authgear/Authgear.kt [hunk 1]
```diff
@@ -71,33 +71,13 @@ constructor(
...
-        )
-    ) {
-        core.addOnSessionStateChangedListener(listener, handler)
-    }
-
-    @MainThread
-    fun removeOnSessionStateChangedListener(listener: OnSessionStateChangedListener) {
-        core.removeOnSessionStateChangedListener(listener)
-    }
+    var delegate: AuthgearDelegate?
```
Comment (kiootic) [inline sdk/src/main/java/com/oursky/authgear/Authgear.kt:74]: I think we should keep the `@MainThread` on getter & setter?

File: sdk/src/main/java/com/oursky/authgear/AuthgearCore.kt [hunk 2]
```diff
@@ -263,30 +261,12 @@ internal class AuthgearCore(
...
-            Looper.getMainLooper()
-        )
-    ) {
-        requireIsMainThread()
-        onSessionStateChangedListeners.add(ListenerPair(listener, handler))
-    }
-
-    @MainThread
-    fun removeOnSessionStateChangedListener(listener: OnSessionStateChangedListener) {
-        requireIsMainThread()
```
Comment (kiootic) [inline sdk/src/main/java/com/oursky/authgear/AuthgearCore.kt:279]: Should we keep this call `requireIsMainThread`?

File: sdk/src/main/java/com/oursky/authgear/AuthgearCore.kt [hunk 3]
```diff
@@ -263,30 +261,12 @@ internal class AuthgearCore(
...
-    }
-
     private fun updateSessionState(state: SessionState, reason: SessionStateChangeReason) {
         // TODO: Add re-entry detection
         sessionState = state
-        onSessionStateChangedListeners.forEach {
-            it.handler.post {
-                it.listener.onSessionStateChanged(authgear, reason)
-            }
+        val handler = Handler(Looper.getMainLooper())
```
Comment (kiootic) [inline sdk/src/main/java/com/oursky/authgear/AuthgearCore.kt:267]: We can delete `ListenerPair.kt` now.

**Comment by louischan-oursky** [comment]
Updated.

**Review by kiootic — APPROVED** [review]

**Pull request merged**